src/routes/publish.js

```diff
@@ -9,23 +9,33 @@
 import Router from '@koa/router';
 import Stream from 'stream';
 import { promisify } from 'util';
+import crypto from 'crypto';
 
 import { getLogger } from '../log';
 import { config } from '../config';
 import { create as gcsStorageCreate } from '../logic/gcs';
 import * as Jwt from '../logic/jwt';
 import {
-  HasherPassThrough,
   LengthCheckerPassThrough,
-  Concatenator,
   CheapJsonChecker,
   forwardErrors,
   GunzipWrapper,
 } from '../utils/streams';
+import { toBase64url } from '../utils/base64url';
 import { PayloadTooLargeError } from '../utils/errors';
 
 const MAX_BODY_LENGTH = 50 * 1024 * 1024; // 50MB
 
+async function generateTokenForProfile(): Promise<string> {
+  const randomBytes = promisify(crypto.randomBytes);
```

Comment:
We use the async version of randomBytes (the one taking a callback) so that it's asynchronous and doesn't block the thread.

Reply:
nit: Should we hoist the promisify call out of the function scope so it's only done once?